synapse/handlers/message.py

```diff
@@ -761,8 +761,10 @@ async def _is_exempt_from_privacy_policy(
     async def _is_server_notices_room(self, room_id: str) -> bool:
         if self.config.servernotices.server_notices_mxid is None:
             return False
-        user_ids = await self.store.get_users_in_room(room_id)
-        return self.config.servernotices.server_notices_mxid in user_ids
+        is_server_notices_room = await self.store.check_local_user_in_room(
+            user_id=self.config.servernotices.server_notices_mxid, room_id=room_id
+        )
+        return is_server_notices_room
```

Comment:
There are several of these `is_server_notices_room` kind of functions but this PR doesn't deduplicate them.